PageViewController.swift

```diff
@@ -0,0 +1,211 @@
+        
+        let closeButton = UIButton()
+        closeButton.setTitle("X", for: .normal)
+        closeButton.setTitleColor(.gray, for: .normal)
+        closeButton.frame = CGRect(x: 30, y: 100, width: 50, height: 30)
+        view.addSubview(closeButton)
+        closeButton.addTarget(self, action: #selector(self.close(_:)), for: .touchUpInside)
+        closeButton.translatesAutoresizingMaskIntoConstraints = false
+        closeButton.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 24).isActive = true
+        closeButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        
+        
+        getStarted.setImage(UIImage(named: "getStarted"), for: .normal)
+        getStarted.frame = CGRect(x: 30, y: 100, width: 186, height: 43)
+        view.addSubview(getStarted)
+        getStarted.addTarget(self, action: #selector(self.close(_:)), for: .touchUpInside)
+        getStarted.translatesAutoresizingMaskIntoConstraints = false
+        getStarted.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+        getStarted.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -130).isActive = true
+        getStarted.isHidden = true
+        getStarted.layer.opacity = 0
+        
+    }
+    
+    @objc func close(_: UIButton) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+
+    @objc func skipClicked(_: UIButton) {
+        
+        for i in currentIndex...pages.count-1 {
+            setViewControllers([pages[i]], direction: .forward, animated: true, completion: nil)
+        }
+        pageControl.currentPage = pages.count - 1
+        
+    }
+    
+    @objc func nextClicked(_: UIButton){
+        print("Nexting now")
+        moveToNextPage()
+    }
+    
+    func moveToNextPage() {
+        
+        if currentIndex != pages.count - 1 {
+            setViewControllers([pages[currentIndex + 1]], direction: .forward, animated: true, completion: nil)
+        }
+        pageControl.currentPage += 1
+        
+    }
+    
+    func configurePageControl() {
+        let initialPage = 0
+        let page1 = StudyPactOnboardingViewController()
+        let page2 = StudyPactOnboarding2ViewController()
+        let page3 = StudyPactOnboarding3ViewController()
+        
+        // add the individual viewControllers to the pageViewController
+        self.pages.append(page1)
+        self.pages.append(page2)
+        self.pages.append(page3)
+        setViewControllers([pages[initialPage]], direction: .forward, animated: true, completion: nil)
+        
+        // pageControl
+        self.pageControl.frame = CGRect(x: 0, y: 0, width: 100, height: 100)
+        self.pageControl.currentPageIndicatorTintColor = UIColor.black
+        self.pageControl.pageIndicatorTintColor = UIColor.lightGray
```

Comment:
figma seems to have white for non-current page circle and blue for current page circle. not a big deal though if there's a problem with doing it that way